### PR TITLE
Add initial structure for coding guidelines subcommittee

### PR DESCRIPTION
**Pull request by PLeVasseur**
Hey folks :wave: 

I'm opening a PR to put some initial thoughts behind the coding guidelines subcommittee based on our kick-off at RustConf.

What I'd like help with:

* If you're interested in becoming a member and not on the list, please sound off here in the PR. The process for becoming a member of the consortium and of subcommittees is still being formulated and I don't want to miss anyone.
* If your name is listed under `members.md` and you **no longer wish to be involved**, please let me know. I know life can get busy :grin: 
* If you're aware of other resources we can highlight in the section on the state of Rust coding guidelines for safety-critical that'd be super helpful to give a shout out here.
  * Or you can message me privately on Zulip if you have a company-specific document you think we could adopt from. We'll need to ensure that any potential IP is contributed.
* If you have some additional thoughts about what our mission statement should be or any other part of the PR, feel free to highlight what you'd like to add / delete / modify

## Next Steps

I plan to send a message to the topic on Zulip for Coding Guidelines asking for review as well as an email to the email list. I'll also come up with some way for us all to vote on some day and time to have a recurring meeting to get started.

## Add availability to the LettuceMeet

Some folks may not be on the Rust Safety-Critical Consortium mailing list and not gotten the notice. I'm looking to find when we can hold our every-two-weeks sync on our goals as a subcommittee.

Please add your regular availability to this LettuceMeet:
https://lettucemeet.com/l/A2eBY

_I'll considered submissions closed on 2024/10/08 at 6pm Eastern in the US._

I'll then tally entries and look for the most common ground to hold the meeting.

## PR Contents

* README.md with the basics of role and how to get in contact
* members.md with current membership as of Sep 10 2024 RustConf kick-off
* mission-statement.md with initial mission statement

### Update `members.md`

- [x] Swap Erik Jonkers for Marc Schoolderman 

### Existing materials section in `mission-statement.md`

- [x] Add the [outcome](https://www.autosar.org/fileadmin/standards/R23-11/AP/AUTOSAR_AP_EXP_ARARustApplications.pdf) of the Rust API for AUTOSAR ARA applications in Rust

### Update `members.md`

If I recall correctly there will be some sort of lightweight process to join up. It may be as simple as what we've done here! But I'd like to discuss in the first meeting what this looks like to make sure we're aligned.

- [ ] Add Munawar Hafiz
- [ ] Add Andrew Davies
- [ ] Add Daniel Glazman
- [ ] Add Sarah Dietrich
- [ ] Add Andrew Herridge
- [ ] Add Lukas Elmer
- [ ] Add Jonas Wolf

### CODE REVIEW

**Comment by iofbd** [comment]
Requesting to be added to the coding guidelines sub-committee. I have been in the mailing lists, but have not been included as a member. 
 
To be added to members.md file

```
+ - Munawar Hafiz
```

Short Bio:
I am Munawar Hafiz, founder and CEO of OpenRefactory, a startup based in Bay Area that builds better static analysis tools that actually find important bugs and do not generate a lot of false warnings (i.e., what the static analysis tools wanted to do, but are not doing now). We are building a Rust tool that will be released (Alpha) in December 2024. List of bugs that we are concentrating on now are listed here: https://fixers.docs.openrefactory.com/rust-1.0-fixer-list

**Comment by openrefactorymunawar** [comment]
Requesting to be added to the coding guidelines sub-committee. I have been in the mailing lists, but have not been included as a member. 
 
To be added to members.md file

```
+ - Munawar Hafiz
```

Short Bio:
I am Munawar Hafiz, founder and CEO of OpenRefactory, a startup based in Bay Area that builds better static analysis tools that actually find important bugs and do not generate a lot of false warnings (i.e., what the static analysis tools wanted to do, but are not doing now). We are building a Rust tool that will be released (Alpha) in December 2024. List of bugs that we are concentrating on now are listed here: https://fixers.docs.openrefactory.com/rust-1.0-fixer-list

File: subcommittee/coding-guidelines/mission-statement.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Mission Statement - Coding Guidelines Subcommittee 
+
+Develop a community-vetted coding guideline for Rust in safety-critical applications. A living document that is updated as features are added to Rust and we learn more by doing.
+
+The following are relevant safety-critical coding guidelines. We are also interested in adopting knowledge obtained that may currently be in company-specific documents and incorporating these into the coding guideline.
```
Comment (jmqd) [inline subcommittee/coding-guidelines/mission-statement.md:5]: I was sort of thinking about this -- do we want to start with generalized coding guidelines for Rust, and then specialize into safety-critical guidelines (as two separate artifacts)? Or do we want to start from the outset focused only on a set of safety-critical focused guidelines? On the one hand, focusing seems desirable, but I believe some folks believed that we needed both. Just opening a discussion to see if anyone has thoughts here.

**Comment by jmqd** [comment]
Thanks for putting this up Pete!

A few things I'm thinking about (for the future, just dumping here).

- I'd like to see our rules to be mechanized, e.g. `TOML` data that gets generated into a document or similar.
- For metadata on each revision of the guidelines itself, to your point of updating with each version of Rust, we should have a field like "updated as of Rust 1.80", etc. To show the last time we did an audit of the latest release.
- I prefer to focus on analyzable rules.

**Comment by NinkasiUK** [comment]
I am Andrew Davies and on the mailing list and would like to take part in the coding standards development. I have 30+ years of developing high integrity systems using SPARK Ada so keen to see if a coding standard can be developed that is comparable to SPARK Ada.

File: subcommittee/coding-guidelines/mission-statement.md (hunk 1 above)
Comment (PLeVasseur) [reply subcommittee/coding-guidelines/mission-statement.md:5]: Yeah, I hear you. IMHO these kinds of things are reasonable topics to bring to our first meeting to try to scope the work.

For my part, from what I've seen in the community and the desire from some in automotive I'd be interested in picking `unsafe` usage as a reasonable starting point to ship to get some momentum behind the group.

File: subcommittee/coding-guidelines/members.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+- Walter Pearce
+- Joel Marcey
+- Alex Celeste
+- Florian Gilcher
+- Pete LeVasseur (Initial Chair)
+- Erik Jonkers
+- Jordan McQueen
+- Christof Petig
+- Alex Vochescu
```
Comment (therealglazou) [inline subcommittee/coding-guidelines/members.md:11]: You can add me to that list; Daniel Glazman (Thales Group)

Comment (sarahkarolina) [reply subcommittee/coding-guidelines/members.md:11]: Please add me as well: Sarah Dietrich (Festo)

**Comment by PLeVasseur** [comment]
To anyone that's jumping in on this PR to say they'd like to join, please reference the PR summary. I've added a LettuceMeet you can add your availability for for our sync meeting!

_Please do so by tomorrow 2024/10/08 by 6pm Eastern in the US._

**Comment by ah801** [comment]
Please add me to the list: Andrew Herridge (Thales Group), thank you.

File: subcommittee/coding-guidelines/members.md (hunk 2 above)
Comment (lukaselmer) [reply subcommittee/coding-guidelines/members.md:11]: Me as well: Lukas Elmer (Siemens)

Comment (jonasthewolf) [reply subcommittee/coding-guidelines/members.md:11]: Please also add me, Jonas Wolf (Vector Informatik)

Comment (JoelMarcey) [reply subcommittee/coding-guidelines/members.md:11]: Also add Marc Schoolderman (Tweede Golf)

**Comment by PLeVasseur** [comment]
Hi there folks :wave: I have noted those that would like to join to add in a follow-up PR. There was some discussion over on the Zulip about designating between "producers" vs "pure listeners" like what's done in some other workgroups.

I'd like to table adding more folks to the `members.md` till we have a bit better idea of what the lightweight process to join looks like. Might be as simple as what we've done here, but just want to review once with folks in the meeting.

**Review by JoelMarcey — APPROVED** [review]
This lgtm to start. We can make tweaks as we go along.

**Pull request merged**